rcl/src/rcl/logging.c

```diff
@@ -188,7 +191,9 @@ rcl_logging_ext_lib_output_handler(
   };
 
   va_list args_clone;
-  va_copy(args_clone, *args);
+  // The args are initialized, but clang-tidy cannot tell.
+  // It may be related to this bug: https://bugs.llvm.org/show_bug.cgi?id=41311
+  va_copy(args_clone, *args);  // NOLINT(clang-analyzer-valist.Uninitialized)
```

Comment:
It's not obvious to me that `args` is initialized. IIUC, they come from this public function and there's no guard for a null value:

https://github.com/ros2/rcl/blob/3a07bef352ff82c2502b1f0502930041d756c064/rcl/src/rcl/logging.c#L155

I think we're primarily relying on `rcutils` to call the function [here](https://github.com/ros2/rcutils/blob/67095577eaf39977f30f7db1ae92f7b0b2ec48d0/src/logging.c#L558) (and `args` is not null), but there's nothing stopping another user from calling `rcl_logging_multiple_output_handler` with a null value.

Reply:
True, but the place that this function is being used, we ensure the args are not-null and initialized (different things).

We can check for null, but that doesn't guard against uninitialized, which would be something like this:

```c
va_list args;  // uninitialized
rcl_logging_multiple_output_handler(..., &args);
```

Reply:
Understood. I agree that `args` is initialized the way we are calling this function.

Reply:
Yeah, so for this function, there's nothing useful for me to do if `args` is `nullptr` since it returns `void` and there's no access to any error handler here. This function is local the `.c` file so I think we just ensure it's always called with non-nullptr. If it were C++ I would say we should use a reference, but because it's not I'd say we just document it and leave it as is.